src/python/espressomd/particle_data.pyx

```diff
@@ -1041,16 +1042,17 @@ cdef class ParticleHandle:
     IF ROTATION:
         property rotation:
             """
-            Switches the particle's rotational degrees of freedom in the Cartesian axes in the body-fixed frame
-            The content of the torque and omega variables are meaningless, for the co-ordinates for which
+            Switches the particle's rotational degrees of freedom in the
+            Cartesian axes in the body-fixed frame. The content of the torque
+            and omega variables are meaningless for the co-ordinates for which
             rotation is disabled.
 
             The default is not to integrate any rotational degrees of freedom.
 
-            rotation : (int,int,int)
+            rotation : list of :obj:`int`
```

Comment:
Should the dimension of the list of `int`s be indicated?

Reply:
The dimension is almost never written for arrays, in most places I saw simply ```array_like :obj:`int` ```. The numpydoc style guide shows examples of both `array_like` and `list of` but none with the dimensions: https://numpydoc.readthedocs.io/en/latest/format.html#sections Even in Doxygen the size of C arrays is never mentioned. I think it would be more helpful to write the dimensions, but keeping that information up-to-date everytime the code changes is probably not possible.

Reply:
At least in obvious cases like things that are and will stay 3d, I would add the dimension. The information that some kind of `array_like`/`list of` is expected without dimension could be rather non-intuitive/confusing to plain ('non programmer') users of espresso, I imagine...

Reply:
Also, for `numpy` `array_like` inputs usually can have an arbitrary dimension or at least length. That does not hold for espresso...

Reply:
I agree that stating the dimension would be more helpful, yet we don't do it (torque_lab, rinertia, omega_body, etc. don't state their dimension) and we probably can't maintain it in the long term (we could re-purpose `check_type_or_throw_except()` to automatically inject the dimension in the docstring, if we don't mind the additional complexity...) and we probably won't agree on which properties/arguments are obvious candidates for stating their dimension.

My impression is that `array_like` is used as a synonym of `list of` in espresso. Shapes for example use `array_like` while particle properties use `list of`. I think it's a historical artifact, because now we accept 3-tuples, 3-lists and 3-np.arrays interchangeably thanks to implicit casts. If the user makes a mistake, an error message is printed with the actual dimension that is expected.

Reply:
offline discussion: @pkreissl will take care of this in a new PR after this one is merged.